src/twisted/python/compat.py

```diff
@@ -37,7 +37,7 @@
 from io import StringIO as NativeStringIO
 from io import TextIOBase
 from sys import intern
-from types import MethodType as _MethodType
+from types import FrameType, MethodType as _MethodType
```

Comment:
Should all imports be prefixed with '_' or is '_MethodType' special?
It would be safer to move to `_foo.py` files with `foo.py` importing the public symbols if we are trying to keep the namespace cleaner than "look at `__all__`", no?

Reply:
you can use https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-reexport

Reply:
Neat. That does seem useful when running `mypy`. The concern here (and I assume the reason for the `_`-prefixed  `_MethodType`) is that symbols in the module's namespace may be accessed by client applications, and removing them arguably requires a deprecation.

Some time ago on a project at work, @glyph started creating separate implementation files and public files that imported only the public symbols, which makes it necessary to be explicit about what is exported, as opposed to `__all__`, which is sort of advisory, and I always do that now.

This is a bit of a rathole, but I'm really just wondering if `_MethodType` suggests something I'm not aware of.

Reply:
My current attempt at a solution to this problem is https://pypi.org/project/publication/ but it doesn't really interact sensibly with mypy yet.

Reply:
OK let's put a pin in that for now.